File: scala/matematica/3096.scala
import scala.math._

object Main extends App {
  def kamenetsky(n: Double): Int = {
    if (n < 0) {
      0
    } else if (n <= 1) {
      1
    } else {
      val x = ((n * log10(n / E)) + (log10(2 * Pi * n) / 2.0))
      math.floor(x).toInt + 1
    }
  }

  val n = scala.io.StdIn.readDouble()
  println(kamenetsky(n))
}

File: c#/matematica/3096.cs
using System;

class Program
{
    static int Kamenetsky(double n)
    {
        if (n < 0)
        {
            return 0;
        }

        if (n <= 1)
        {
            return 1;
        }

        double x = ((n * Math.Log10(n / Math.E)) +
                    (Math.Log10(2 * Math.PI * n) / 2.0));

        return (int)(Math.Floor(x) + 1);
    }

    static void Main()
    {
        double n;
        if (double.TryParse(Console.ReadLine(), out n))
        {
            Console.WriteLine(Kamenetsky(n));
        }
    }
}
